# PAD - Praca domowa 2
## Jan Solarz s26342

### Zadanie 1

Zaczynamy od wgrania ramki danych

In [35]:
import pandas as pd
PATH = "Downloads\PAD_02_PD.csv"

In [36]:
df = pd.read_csv(PATH, sep=";")
df.shape

(800, 8)

In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 800 entries, 0 to 799
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   ID          800 non-null    int64  
 1   Country     793 non-null    object 
 2   owns_car    798 non-null    float64
 3   owns_TV     798 non-null    float64
 4   owns_house  794 non-null    float64
 5   owns_Phone  800 non-null    int64  
 6   gender      795 non-null    object 
 7   Age         800 non-null    int64  
dtypes: float64(3), int64(3), object(2)
memory usage: 50.1+ KB


Widoczne są braki danych w zmiennych

In [38]:
df['Country'].value_counts(dropna=False)
#co z wartosciami NA

A      328
B      201
C      186
D       78
NaN      7
Name: Country, dtype: int64

Powyżej wskazano na liczebność każdej z wartości w zmiennej COUNTRY. 7 wartości jest NaN.

In [39]:
df=df.dropna()

In [40]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 778 entries, 0 to 799
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   ID          778 non-null    int64  
 1   Country     778 non-null    object 
 2   owns_car    778 non-null    float64
 3   owns_TV     778 non-null    float64
 4   owns_house  778 non-null    float64
 5   owns_Phone  778 non-null    int64  
 6   gender      778 non-null    object 
 7   Age         778 non-null    int64  
dtypes: float64(3), int64(3), object(2)
memory usage: 54.7+ KB


Ramka danych df została wyczyszczona z wartości NA, w wyniku czego pozostało 778 kompletnych obserwacji.

### Zadanie 2

Tworzymy zmienną "Number_of_goods" jako suma wszystkich dóbr dla poszczególnej osoby.

In [41]:
df["Number_of_goods"] = df["owns_car"]+df["owns_TV"]+df["owns_house"]+df["owns_Phone"]

In [42]:
df[5:20]

ID Country  owns_car  owns_TV  owns_house  owns_Phone gender  Age  \
5    6       A       0.0      0.0         0.0           1      M   48   
6    7       D       1.0      1.0         1.0           1      K   58   
7    8       C       0.0      1.0         0.0           1      K   59   
8    9       A       0.0      0.0         0.0           1      M   61   
9   10       A       0.0      0.0         0.0           1      M   48   
10  11       D       1.0      1.0         1.0           1      K   34   
11  12       B       0.0      1.0         0.0           1      K   42   
12  13       B       0.0      1.0         0.0           1      K   37   
13  14       A       0.0      0.0         0.0           1      M   51   
14  15       D       1.0      1.0         1.0           1      K   42   
15  16       D       1.0      1.0         1.0           1      K   29   
16  17       A       0.0      0.0         0.0           0      M   19   
17  18       A       0.0      0.0         0.0           1      M   30   
18  19       A       0.0      0.0         0.0           0      M   38   
19  20       D       1.0      1.0         1.0           1      K   47   

    Number_of_goods  
5               1.0  
6               4.0  
7               2.0  
8               1.0  
9               1.0  
10              4.0  
11              2.0  
12              2.0  
13              1.0  
14              4.0  
15              4.0  
16              0.0  
17              1.0  
18              0.0  
19              4.0

### Zadanie 3

In [43]:
o=round(df.groupby('gender')['Number_of_goods'].mean(),3)
o

gender
K    2.799
M    0.886
Name: Number_of_goods, dtype: float64

In [44]:
print("Średnia posiadanych dóbr u kobiet wynosi:",o[0], ", u mężczyzn:",o[1])

Średnia posiadanych dóbr u kobiet wynosi: 2.799 , u mężczyzn: 0.886


### Zadanie 4

Tworzymy podsumowującą ramkę danych pogrupowaną po unikalnych krajach.

In [59]:
a = df["Country"].unique()
b=df.groupby('Country')['Age'].min()
c=round((df.groupby('Country')['Number_of_goods'].mean()),2)


In [60]:
d=[]
for i in a:
    d1 = round(len(df[(df['Country'] == i )& (df['gender'] == "K")])/len(df[df['Country'] ==i])*100,2)
    d.append(d1)
d=list(map("{}%".format, d))
  
    

In [62]:
pd.DataFrame({'Unique Country':list(a),"Avg Goods":list(c),'Min Age':list(b),'Percentage of wmns':list(d)})

Unique Country  Avg Goods  Min Age Percentage of wmns
0              A       0.86       18               0.0%
1              B       1.46       18             63.78%
2              C       3.03       18             100.0%
3              D       4.00       18             100.0%

Największa ilość posiadanych dóbr występuje w kraju D na poziomie 4.0. We wszystkich krajach minimalny wiek wynosi 18 lat. W kraju A nie występują kobiety, natomiast w krajach C i D brak jest mężczyzn w danych. 